File: challenges/exercism/javascript/easy/raindrops.js
'use strict';

/**
 * Converts num into raindrop sounds based on divisibility:
 * - if divisible by 3, add "Pling" to the result;
 * - if divisible by 5, add "Plang" to the result;
 * - if divisible by 7, add "Plong" to the result;
 * - if not divisible by 3, 5, or 7, return num as a string.
 */
export const convert = (num) => {
  let result = '';
  if (!(num % 3)) result += 'Pling';
  if (!(num % 5)) result += 'Plang';
  if (!(num % 7)) result += 'Plong';

  return result.length > 0 ? result : String(num);
};

console.assert(convert(1) === '1');

File: challenges/exercism/rust/src/easy/raindrops.rs
pub fn raindrops(n: u32) -> String {
    let mut result = String::new();
    if n % 3 == 0 {
        result.push_str("Pling")
    };
    if n % 5 == 0 {
        result.push_str("Plang")
    };
    if n % 7 == 0 {
        result.push_str("Plong")
    };
    if n % 3 != 0 && n % 5 != 0 && n % 7 != 0 {
        result.push_str(&n.to_string())
    };

    result
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn raindrops_15() {
        assert!(raindrops(15) == "PlingPlang");
    }

    #[test]
    fn raindrops_6() {
        assert!(raindrops(6) == "Pling");
    }

    #[test]
    fn raindrops_8() {
        assert!(raindrops(8) == "8");
    }

    #[test]
    fn raindrops_105() {
        assert!(raindrops(105) == "PlingPlangPlong");
    }

    #[test]
    fn raindrops_3125() {
        assert!(raindrops(3125) == "Plang");
    }
}
